### DISCUSSION TITLE
The member 'ProcessedTableManager' can only be used within its package.

### DISCUSSION THREAD

**Discussion by Foxsterc** [simolus3/drift#3214] [General]
Hi,

Weird one this.

drift: ^2.20.2
drift_dev: ^2.20.3

Flutter 3.24.2 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 4cf269e36d (8 days ago) • 2024-09-03 14:30:00 -0700
Engine • revision a6bd3f1de1
Tools • Dart 3.5.2 • DevTools 2.37.2

I seem to be getting the following warnings in the generated code file.

The member 'ProcessedTableManager' can only be used within its package.

and

The '!' will have no effect because the receiver can't be null.
Try removing the '!' operator.

<img width="654" alt="image" src="https://github.com/user-attachments/assets/0b8550b3-819e-47c2-8efb-c86cd2beaef8">

``
<img width="661" alt="image" src="https://github.com/user-attachments/assets/a24ad18d-418c-4a0d-a248-d3a98a582b2d">

I have tried changing the packages, removing the lock files downgrading Flutter, re-generating the code. Any ideas why this might be?

Thanks,

**Comment by dickermoshe** [comment]
We ignore lints in `*.g.dart` files so we didn't catch this.
@simolus3, Being that by default `*.g.dart` files are linted, should we add ignores ot the top of generated files?

**Reply by Foxsterc** [reply]
Ok thanks, I'll just ignore them. It must be a recent update somewhere maybe on the flutter_lints as this is the first I've seen this.

**Reply by dickermoshe** [reply]
Nope, it's just been added recently to drift in 2.18

**Reply by simolus3** [reply]
> @simolus3, Being that by default `*.g.dart` files are linted, should we add ignores ot the top of generated files?

We have a `// ignore_for_file: type=lint` at the top of generated code. I wonder if the problem is that users are upgrading this lint to a warning, causing these issues? If there's a more generic comment that disables more analysis hints, IMO we should consider applying it.

**Reply by Foxsterc** [reply]
@simolus3 I've not done anything with the lints, that I know of.  // ignore_for_file: type=lint is present at the top

**Reply by dickermoshe** [reply]
can you post your `analysis_options.yaml` file?

**Reply by Foxsterc** [reply]
@dickermoshe it just has

include: package:flutter_lints/flutter.yaml

**Reply by simolus3** [reply]
@dickermoshe Given how many people seem to run into this, should we perhaps just remove the annotation and add a comment explaining that the class should only be used in generated code instead?

**Reply by dickermoshe** [reply]
PR: https://github.com/simolus3/drift/pull/3233

**Reply by dickermoshe** [reply]
Merged, should be fixed on next release
